modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxState.java

```diff
@@ -21,7 +21,6 @@
  * Transaction state.
  */
 public enum TxState {
-    PENDING,
```

Comment:
If we are getting rid of pending state, I think, we should to modify TxManagerImpl#changeState method in order of it doesn't check the state null before CAS. The null is the certain state here, it has to check as others.

Reply:
fixed